src/code42cli/output_formats.py

```diff
@@ -30,15 +46,33 @@ def output_format(_, __, value):
 )
 
 
+extraction_format_option = click.option(
+    "-f",
+    "--format",
+    type=click.Choice(OutputFormat(), case_sensitive=False),
+    help="The output format of the result. Defaults to table format.",
+    callback=extraction_output_format,
+)
+
+
+def to_dynamic_csv(output, header):
```

Comment:
Created a separate function that creates `csv` output with 'csv' module. As I did not want to change existing function where we had flexibility to set custom column headers. I couldn't figure out how that would be achieved with `csv` module which takes `fieldnames` as column header by default. Also note that header is based on keys found in first record of `events`.

Reply:
Custom headers look nice but I am wondering if it is just simply better to use response keys themselves.

They can vary though - `legalHoldUid` is `Matter ID`.

`Matter ID` looks nice but now we are maintaining a mapping _and_ obfuscating results.